ChatViewController.m

```diff
@@ -43,13 +44,21 @@ - (void) viewDidLoad {
         // Do any additional setup after loading the view.
 }
 
+- (void) viewWillAppear:(BOOL)animated {
+    [self setupNavBar];
+}
+
 - (void) didReceiveMemoryWarning {
     [super didReceiveMemoryWarning];
     // Dispose of any resources that can be recreated.
 }
 
 - (IBAction) onBackTapped:(id)sender {
-    [self dismissViewControllerAnimated:YES completion:nil];
+    [self.navigationController popViewControllerAnimated:YES];
```

Comment:
Because with a push navigation, popViewControllerAnimated is the equivalent of dismissViewController